pkg/apis/core/validation/validation.go

```diff
@@ -4601,12 +4599,11 @@ func ValidatePodUpdate(newPod, oldPod *core.Pod, opts PodValidationOptions) fiel
 		mungedPodSpec.TerminationGracePeriodSeconds = oldPod.Spec.TerminationGracePeriodSeconds // +k8s:verify-mutation:reason=clone
 	}
 
-	if !apiequality.Semantic.DeepEqual(mungedPodSpec, oldPod.Spec) {
-		// This diff isn't perfect, but it's a helluva lot better an "I'm not going to tell you what the difference is".
-		// TODO: Pinpoint the specific field that causes the invalid error after we have strategic merge diff
-		specDiff := cmp.Diff(oldPod.Spec, mungedPodSpec)
-		allErrs = append(allErrs, field.Forbidden(specPath, fmt.Sprintf("pod updates may not change fields other than `spec.containers[*].image`, `spec.initContainers[*].image`, `spec.activeDeadlineSeconds`, `spec.tolerations` (only additions to existing tolerations) or `spec.terminationGracePeriodSeconds` (allow it to be set to 1 if it was previously negative)\n%v", specDiff)))
```

Comment:
This one is unfortunate, I'll leave it up to Jordan, I would maybe leave these as they are tbh since there's more value in the specific error message than in simplifying the code just a little bit.

Reply:
I see, the original code is restored here（